verilog/tools/ls/autoexpand.cc

```diff
@@ -1387,10 +1387,24 @@ std::vector<TextEdit> ConvertAutoExpansionsToFormattedTextEdits(
   // TextEdits.
   std::string text;
   auto begin = text_structure.Contents().begin();
-  for (const AutoExpander::Expansion &expansion : expansions) {
-    text.append(begin, expansion.replaced_span.begin());
-    text.append(expansion.new_text);
-    begin = expansion.replaced_span.end();
+  for (auto it = expansions.begin(), next = it; it != expansions.end();
```

Comment:
this new snippet looks like it wants to be a separate function with a name that explains what it does.